src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseExceptionThrowHelpersTests.cs

```diff
@@ -0,0 +1,1078 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.UseExceptionThrowHelpers,
+    Microsoft.NetCore.Analyzers.Runtime.UseExceptionThrowHelpersFixer>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.UseExceptionThrowHelpers,
+    Microsoft.NetCore.Analyzers.Runtime.UseExceptionThrowHelpersFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class UseExceptionThrowHelpersTests
+    {
+        [Fact]
+        public async Task ArgumentNullExceptionThrowIfNull_DoesntExist_NoDiagnostics()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+namespace System
+{
+    public class ArgumentNullException : Exception
+    {
+        public ArgumentNullException(string paramName) { }
+        public ArgumentNullException(string paramName, string message) { }
+    }
+}
+
+class C
+{
+    void M(string arg)
+    {
+        if (arg is null)
+            throw new ArgumentNullException(nameof(arg));
+
+        if (arg == null)
+            throw new ArgumentNullException(nameof(arg));
+
+        if (null == arg)
+            throw new ArgumentNullException(nameof(arg));
+
+        if (arg is null)
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }
+
+        if (arg == null)
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }
+
+        if (null == arg)
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }
+    }
+
+    string this[string name]
+    {
+        get
+        {
+            if (name is null)
+                throw new ArgumentNullException(nameof(name));
+            return name;
+        }
+    }
+}
+");
+        }
+
+        [Fact]
+        public async Task ArgumentNullExceptionThrowIfNull_FixAppliesAsAppropriate()
+        {
+            await new VerifyCS.Test()
+            {
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                TestCode =
+@"
+using System;
+
+namespace System
+{
+    public class ArgumentNullException : Exception
+    {
+        public ArgumentNullException() { }
+        public ArgumentNullException(string paramName) { }
+        public ArgumentNullException(string paramName, string message) { }
+        public ArgumentNullException(string paramName, Exception innerException) { }
+        public static void ThrowIfNull(object argument, string name = null) { }           
+    }
+}
```

Comment:
Can we get rid of this and let the test run on .NET 7 with `ReferenceAssemblies = ReferenceAssemblies.Net.Net70`?

Reply:
We could do that for some of them, like this one.  Others are new in .NET 8.  Seemed like the most consistent thing was to always specify them and not require a particular set of reference assemblies.